<?xml version="1.0" ?>
<launch>
  <arg name="use_present" default="true" />
  <param name="robot_description" command="$(find xacro)/xacro.py '$(find rh_p12_rn_description)/urdf/thormang3.xacro'" />
    
  <!-- send joint values -->
  <node pkg="joint_state_publisher" type="joint_state_publisher" name="robotis_joint_state_publisher">
    <param name="use_gui" value="FALSE"/>
    <param name="rate" value="30"/>
    <remap from="/joint_states" to="/robotis/rh_p12_rn/joint_states" />
    <rosparam param="source_list" if="$(arg use_present)">["robotis/rh_p12_rn/present_joint_states"]</rosparam>
    <rosparam param="source_list" unless="$(arg use_present)">["robotis/rh_p12_rn/goal_joint_states"]</rosparam>
  </node>

  <!-- Combine joint values to TF-->
  <node pkg="robot_state_publisher" type="state_publisher" name="robot_state_publisher">
    <remap from="/joint_states" to="/robotis/rh_p12_rn/joint_states"/>
  </node>

  <!-- Gripper Pub-->
  <node pkg="rh_p12_rn_description" type="rviz_rh_pub" name="rviz_rh_pub">
  </node>
</launch>
